src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -14,43 +17,43 @@ namespace Microsoft.VisualStudio.LanguageServices.Experimentation
     [ExportWorkspaceService(typeof(IExperimentationService), ServiceLayer.Host), Shared]
     internal class VisualStudioExperimentationService : ForegroundThreadAffinitizedObject, IExperimentationService
     {
-        private readonly object _experimentationServiceOpt;
-        private readonly MethodInfo _isCachedFlightEnabledInfo;
+        private readonly IAsyncServiceProvider _asyncServiceProvider;
+
+        private Optional<Func<string, bool>> _isCachedFlightEnabled;
```

Comment:
could consider an async lazy.

Reply:
➡️ Since a UI thread dependency is involved, I prefer to avoid the confusion between Roslyn's `AsyncLazy<T>` and the JTF-aware one defined in vs-threading.

Reply:
interesting.  for my own edification, do we think roslyn's Async lazy could not wrap JTF async code?  Or is it more just being paranoid here?  

> and the JTF-aware one defined in vs-threading.

I'd also be fine with this.  It was more about me seeing that there were non-problematic race conditions in teh code below.  AsyncLazy (roslyn or jtf) just seemed like a clean way to expres intent.  but it's not somethign critical at all.

Reply:
> for my own edification, do we think roslyn's Async lazy could not wrap JTF async code?

Yesterday would have been the perfect day to ask this 🎃

Reply:
:D    

👻  `.Result`  🎃 

> I'm honestly not sure

Good enough reason to avoid for now :D

Reply:
> interesting. for my own edification, do we think roslyn's Async lazy could not wrap JTF async code? Or is it more just being paranoid here?

@CyrusNajmabadi My expectation is no, simply because we're using TaskCreationSource behind the scenes. A JTF.Run() on the UI thread that implicitly waits on an AsyncLazy won't know what to let through and what won't.

Reply:
@jasonmalinowski Thanks!